### DISCUSSION TITLE
Where to put  HxMessenger in Blazor 8?

### DISCUSSION THREAD

**Discussion by chobo2** [havit/Havit.Blazor#695] [Q&A]
Hi

I am using blazor 8 server with per component / per page and I tried to follow the instructions from https://havit.blazor.eu/components/HxMessenger

It says to put it in this` <HxMessenger Position="ToastContainerPosition.TopEnd" />` in the main layout.

I did not see any boxes to show up even after I called the messenger class. 

I suspected because of the "render mode" so I went to one of my components that had InteractiveServer mode set and put this in
`<HxMessenger Position="ToastContainerPosition.TopEnd" />`

and it did work. I tired to put a render mode on HxMessenger in MainLayout but it did not like that. I then tried to put it on the top of MainLayout but it did not like it.

**Comment by leonardopugliese** [comment]
Sorry for the delay in the response. In case I haven't solved it or in any case if anyone has the same type of problem, I'll point out the solution that worked for me.

In my case the problem arose when using the new .Net 8.0 template with the new .razor pages for Identity and of course the Havit components.

Here is the workaround [https://github.com/dotnet/aspnetcore/issues/51476](https://github.com/dotnet/aspnetcore/issues/51476)

**Reply by chobo2** [reply]
Hey, I have to look what I did as I think I solved it but your link does not seem to work.

**Reply by leonardopugliese** [reply]
Sorry, fixed
